src/app/workouts/workoutCreate/WorkoutCreate.tsx

```diff
@@ -0,0 +1,142 @@
+// src/pages/WorkoutCreateFromRoutine.tsx
+
+import React, { useEffect, useState } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+import { Workout } from '@/types/entities/Workout';
+import { RoutineExercise } from '@/types/entities/Routine';
+import { createWorkout } from '@/api/workoutApi';
+import { AppRoutes } from '@/types/routes';
+import {
+  Button,
+  Typography,
+  List,
+  ListItem,
+  ListItemText,
+  Divider, Grid,
+} from '@mui/material';
+
+//TODO: to be removed
+import { mockRoutineData } from '@/types/entities/mockRoutineData';
+
+const WorkoutCreate: React.FC = () => {
+  const location = useLocation();
+  const navigate = useNavigate();
+  const [workout, setWorkout] = useState<Workout | null>(null);
+
+  useEffect(() => {
+    // TODO: remove comment when proper data fetching exists
+    // const routineData = location.state?.routineData;
+    const routineData = mockRoutineData; // mocked routineData
```

Comment:
We should mock that data at routine level, so here you are accessing mocked data from state that you are sending in navigate

Reply:
logical
will do